# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [17]:
# number of callbacks for black-sounding names
b_callback = sum(data[data.race=='b'].call)
b_callback

157.0

In [16]:
# number of callbacks for white-sounding names
w_callback = sum(data[data.race == 'w'].call)
w_callback

235.0

In [6]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [10]:
df = data[['race', 'call']]
df.head()

race  call
0    w   0.0
1    w   0.0
2    b   0.0
3    b   0.0
4    w   0.0

In [11]:
# Check if there are any null values in the data. Are are not.
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 2 columns):
race    4870 non-null object
call    4870 non-null float32
dtypes: float32(1), object(1)
memory usage: 95.1+ KB


In [14]:
# Check how the observations are split between white and black.
df.race.value_counts()

b    2435
w    2435
Name: race, dtype: int64

In [48]:
# Check the total split between call backs and non-call backs.
df.call.value_counts()

0.0    4478
1.0     392
Name: call, dtype: int64

In [49]:
# Out of the 392 call backs, how many were for candidates with a white-sounding name versus candidates with
# black-sounding name?
df_pivot = pd.pivot_table(df, values='call', columns='race', aggfunc=np.sum)
df_pivot

race      b      w
call  157.0  235.0

### 1. What test is appropriate for this problem? Does CLT apply?

It is appropriate to use the t-test in this scenario since we don't have the population mean or standard deviation. CLT does apply given we have 4878 observations.

### 2. What are the null and alternate hypotheses?

The null hypothesis is: White and black-sounding names receive the same number of call backs for the same resume.

The altenative hypothesis is: White and black-sounding names receive different number of call backs for the same resume.

### 3. Compute margin of error, confidence interval, and p-value.

In [104]:
# Split the observations between white and black
white = df[df.race == 'w']
black = df[df.race == 'b']

# Average of callbacks
white_mean = np.mean(white.call)
black_mean = np.mean(black.call)

print('Percentage of call back for white-sounding names:', white_mean)
print('Percentage of call back for black-sounding names:', black_mean)
print()

# Difference of means
diff_of_means = white_mean - black_mean
print('The difference in means is:', diff_of_means)
print()

# t-score and p-value
t, p = stats.ttest_ind(white.call, black.call)
print('t-score:', t)
print('p-value:', p)
print()

# Standard Error
SE = np.sqrt(np.std(white.call)**2 / n_white + np.std(black.call)**2 / n_black)

margin_of_error = 1.96 * SE
print('Margin of error:', merror)
print()

# Confidence Interval
conf_int_95 = [diff_of_means - margin_of_error, diff_of_means + margin_of_error]
print('95% confidence internval:', conf_int_95)

Percentage of call back for white-sounding names: 0.09650924056768417
Percentage of call back for black-sounding names: 0.0644763857126236

The difference in means is: 0.03203285485506058

t-score: 4.11470529086
p-value: 3.94080210313e-05

Margin of error: 0.0152584173807

95% confidence internval: [0.016777570469610682, 0.047288139240510473]


### 4. Write a story describing the statistical significance in the context or the original problem.

In this problem, I sought out to find out if white-sounding names and black-sounding names have an impact on callback rates from potential employers. 

From the data this study was able to gather, I'm able to see that on average, white-sounding names get a 9.65% callback rate, while black_sounding names get a 6.45% callback rate, even though the two groups have the same resume. This is alarming as it suggests that white-sounding names get 1.5 more interviews compared to black-sounding names.

In order to see if these results could be random, I ran a two sample t-test. The analysis shows that the name of a person has statistically significant results on whether that person gets a callback from a potential employer.

### 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis indicates that race/name is a factor in callback success, thought it is difficult to say if it is the most important factor. Further investigation is required to verify if race/name is the most important factor in a candidates callback rate. 

I'd be interested to know how the different columns in this dataset can be used to further this analysis. 